### ISSUE TITLE
Jakarta Connections-2.0 TCK JAX-WS dependency failure

### ISSUE THREAD

**Issue by KyleAure** [OpenLiberty/open-liberty#15087]
Jakarta Connection TCK failing with:

```
[11/20/20 13:34:23:004 EST] 00000020 id=00000000 LogService-151-com.ibm.ws.jca.jakarta                        E CWWKE0701E: bundle com.ibm.ws.jca.jakarta:1.0.47.ctstest-liberty20201117-0521 (151)[com.ibm.ws.jca.internal.BootstrapContextImpl(799)] : The activate method has thrown an exception java.lang.NoClassDefFoundError: jakarta/xml/ws/WebServicePermission
	at com.sun.ts.tests.common.connector.whitebox.permissiondd.PermissionDDResourceAdapterImpl.start(PermissionDDResourceAdapterImpl.java:92)
	at com.ibm.ws.jca.internal.BootstrapContextImpl.activate(BootstrapContextImpl.java:348)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
    ... more
Caused by: java.lang.ClassNotFoundException: jakarta.xml.ws.WebServicePermission
	at com.ibm.ws.classloading.internal.AppClassLoader.findClassCommonLibraryClassLoaders(AppClassLoader.java:569)
	at com.ibm.ws.classloading.internal.AppClassLoader.findClass(AppClassLoader.java:291)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at com.ibm.ws.classloading.internal.AppClassLoader.findOrDelegateLoadClass(AppClassLoader.java:547)
	at com.ibm.ws.classloading.internal.AppClassLoader.loadClass(AppClassLoader.java:505)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 86 more
```
I was not under the impression that connectors had a dependency on JAX-WS so I am a little unsure of why the TCK is attempting to load it for connectors testing.

**Comment by KyleAure** [comment]
Upon further investigation it seems that JAX-WS introduced a new permission class in it's 3.0 release named `jakarta.xml.ws.WebServicePermission`
The Connectors TCK is testing to make sure that this permission is handled correctly by JCA Security. 

Doc: https://jakarta.ee/specifications/xml-web-services/3.0/jakarta-xml-ws-spec-3.0.pdf section 5.2.3

Which is resulting in 
```
[11/18/20 15:00:26:260 EST] 00000018 id=00000000 com.ibm.ws.jca.security.internal.SecWorkContextHandler       E J2CA0671E: The WorkManager was unable to associate the provided SecurityContext to the Work instance. 
                                                                                                               com.ibm.websphere.security.WSSecurityException: J2CA0668E: The WorkManager was unable to populate the execution subject with the caller principal or credentials necessary to establish the security context for this work instance.
	at com.ibm.ws.jca.security.internal.SecWorkContextHandler.associate(SecWorkContextHandler.java:151)
	at com.ibm.ws.jca.inbound.security.SecurityInflowContext.taskStarting(SecurityInflowContext.java:119)
	at com.ibm.ws.context.service.serializable.ThreadContextDescriptorImpl.taskStarting(ThreadContextDescriptorImpl.java:405)
	at com.ibm.ws.context.service.serializable.ContextualRunnable.run(ContextualRunnable.java:77)
	at com.ibm.ws.jca.internal.WorkProxy.call(WorkProxy.java:287)
	at com.ibm.ws.jca.internal.WorkManagerImpl.doWork(WorkManagerImpl.java:112)
	at com.ibm.ws.jca.internal.WorkManagerImpl.doWork(WorkManagerImpl.java:83)
	at com.sun.ts.tests.common.connector.whitebox.ContextWork.run(ContextWork.java:99)
	at com.ibm.ws.jca.inbound.security.JCASecurityContextService$1.run(JCASecurityContextService.java:37)
	at com.ibm.ws.jca.inbound.security.JCASecurityContextService$1.run(JCASecurityContextService.java:34)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.ibm.websphere.security.auth.WSSubject.doAs(WSSubject.java:123)
	at com.ibm.websphere.security.auth.WSSubject.doAs(WSSubject.java:90)
	at com.ibm.ws.jca.inbound.security.JCASecurityContextService$2.run(JCASecurityContextService.java:44)
	at com.ibm.ws.jca.inbound.security.JCASecurityContextService$2.run(JCASecurityContextService.java:41)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.ibm.ws.jca.inbound.security.JCASecurityContextService.runInInboundSecurityContext(JCASecurityContextService.java:41)
	at com.ibm.ws.jca.internal.WorkProxy.run(WorkProxy.java:356)
	at com.ibm.ws.context.service.serializable.ContextualRunnable.run(ContextualRunnable.java:79)
	at com.ibm.ws.jca.internal.WorkProxy.call(WorkProxy.java:287)
	at com.ibm.ws.jca.internal.WorkManagerImpl.doWork(WorkManagerImpl.java:112)
	at com.ibm.ws.jca.internal.WorkManagerImpl.doWork(WorkManagerImpl.java:83)
	at com.sun.ts.tests.common.connector.whitebox.mdcomplete.MDCompleteWorkManager.testNestedContext3(MDCompleteWorkManager.java:206)
	at com.sun.ts.tests.common.connector.whitebox.mdcomplete.MDCompleteWorkManager.runTests(MDCompleteWorkManager.java:63)
```

And failing TCK tests when resource adapters are unavailable for lookup. 

See permission.xml settings at:
https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/src/com/sun/ts/tests/common/connector/whitebox/permissiondd/permissions.xml#L20-L28

**Comment by jhanders34** [comment]
This isn't a new permission really though right?  It is just refactored to be jakarta instead of javax.

**Comment by jhanders34** [comment]
Is this just an issue that xmlWS feature is not in the convenience feature yet so the class is not exported to the application classloader?

**Comment by KyleAure** [comment]
Hey @jhanders34,
According to the documentation it does sound like a new permission.  Though the doc could have just been copied from the last release of Java EE.  @neuwerk could confirm this. 

To answer your question, maybe?  I guess it depends on how to TCK is run.  If the TCK is run by enabling the convenience feature (`jakartaee-9.0`) then yes once `xmlWS` is included in that feature than this TCK should succeed.  But, if the TCK is run with only `connectors-2.0` feature than this will always fail as connectors does not have (nor should have) a dependency on `xmlWS`.  Which is why I think it is weird that the TCK has a test with this permission to begin with.

**Comment by jhanders34** [comment]
The TCK runs with the convenience feature and since xmlWS isn't in there yet, that explains why it isn't found as you said.  xmlWS feature does exist so Brian and team could add that to the convenience feature or update the server.xml to have xmlWS-3.0 in it temporarily to see if it helps this particular problem.

**Comment by KyleAure** [comment]
Issue resolved once xmlWS-3.0 was put in jakartaee-9.0 feature.
